File: Day_3/Data_Structures.py
# ELISA LAM | 2 AUGUST 2023

def myFunc(nums, target):
    # Splitting the numbers
    numsList = nums.split()
            
    # Converting the target into an integer
    target = int(target)

    # Loop that checks if it's the same
    for i in range(0, len(numsList), 1):
        for j in range(i, len(numsList), 1): 
            if (i != j):
                if ((int(numsList[i]) + int(numsList[j])) == target):
                    return (i, j) # Ends function
                
    # If there was nothing found:
    return "faulty input / faulty code"

nums = input("Please enter the list: ")
target = input("Please enter the target: ")

output = myFunc(nums, target)

print("FINAL OUTPUT:", output)

File: Day_3/Data_Structures.java
public class Data_Structures {
    public static int myFunc(int[] nums, int target)
    {
        for (int i = 0; i < nums.length; i++)
        {
            for (int j = i; j < nums.length; j++)
            {
                if (i != j)
                {
                    if (nums[i] + nums[j] == target)
                    {
                        System.out.println("(" + Integer.toString(i) + ", " + Integer.toString(j) + ")");
                        return 0;
                    }
                }
            }
        }

        System.out.println("Faulty code / Faulty input");
        return 1;
    }


    public static void main(String[] args) 
    {
        // Enter code here. 
        int[] nums = {2, 7, 11, 15};
        int target = 9;

        myFunc(nums, target);
    }
}